cookbooks/cdo-varnish/test/integration/cloudfront/minitest/test_default.rb

```diff
@@ -0,0 +1,315 @@
+# Chef test suite for CloudFront distribution behavior.
+
+# Chef node attributes as constants
+DASHBOARD_PORT = 8080
+PEGASUS_PORT = 8081
+VARNISH_PORT = 80
+LOCALHOST = 'localhost'
+VARNISH_URL = "#{LOCALHOST}:#{VARNISH_PORT}"
+ENVIRONMENT = 'integration'
+CLOUDFRONT_URL = "https://#{ENVIRONMENT}-studio.code.org"
+
+# Setup the mock-server and ngrok
+puts 'setup'
+NGROK_PID = spawn('/usr/local/ngrok/ngrok start cdo --config=/home/kitchen/.ngrok2/ngrok.yml --log=stdout')
+PID = spawn('cd ~; java -jar mock.jar')
```

Comment:
1. The integration test can run on your laptop, if Test Kitchen and Docker are installed locally. I've documented this in `cdo-varnish/README.md`:

> ## Running Tests
The integration tests run using [Test Kitchen](http://kitchen.ci/).
See `test/cookbooks/varnish_test/README.md` for more details.
>
> To test the cookbook, first make sure Docker is installed and running locally,
then run:
- `chef exec kitchen create` to create the machine image
- `chef exec kitchen converge` to install Chef and converge the cookbook in the
  platform environment
- `chef exec kitchen verify` to run the integration test suite

I can clarify this documentation to add that `test-kitchen`, `kitchen-docker` and `berkshelf` all need to be available to the Ruby environment (either installed globally via `gem install test-kitchen`, provided by Bundler through the provided `Gemfile` after a `bundle install` via `bundle exec kitchen`, or provided by ChefDK via `chef exec kitchen`).

2\. This will only work in the provisioned, isolated environment created by Test Kitchen. The actual `test_default.rb` code is run within the environment via [Busser](https://github.com/test-kitchen/busser) using the [busser-minitest](https://github.com/test-kitchen/busser-minitest) runner plugin, after the cookbooks and runlists specified in `.kitchen.yml` have been applied to the environment. This is all described in the Test Kitchen documentation.

Let me know if you have any suggestions on how to clarify any of these points within the documentation here.